utils/chroma_store.py

```diff
@@ -0,0 +1,159 @@
+"""ChromaDB vector store class."""
+import ipaddress
+from dataclasses import dataclass
+from typing import List, Optional
+
+import chromadb
+from chromadb.api.types import CollectionMetadata, EmbeddingFunction
+from chromadb.config import Settings
+
+MIN_COLLECTION_NAME_LENGTH = 3
+MAX_COLLECTION_NAME_LENGTH = 64
+
+
+@dataclass
+class ValidateCollectionName:
```

Comment:
A better name here might be: `CollectionNameValidationOutcome` as it's more descriptive